sympy/solvers/tests/test_solvers.py

```diff
@@ -404,6 +404,16 @@ def test_linear_system():
 
     assert solve([x + y + z + t, -z - t], x, y, z, t) == {x: -y, z: -t}
 
+    # https://github.com/sympy/sympy/issues/27854
+    m, n = symbols("m n")
```

Comment:
This is just putting the test in a random place.

It looks like `test_issue_3870` is the main test for using `solve` with a `Matrix`. Let's rename that to `test_solve_Matrix` and add this test there.